src/ipfs-core/init.js

```diff
@@ -0,0 +1,98 @@
+const peerId = require('peer-id')
+
+module.exports = (repo, opts, callback) => {
+  opts = opts || {}
+  opts.force = opts.force || false
+  opts.emptyRepo = opts.emptyRepo || false
+  opts.bits = opts.bits || 2048
+
+  // Pre-set config values.
+  var config = {
+    'Addresses': {
+      'Swarm': [
+        '/ip4/0.0.0.0/tcp/4001',
+        '/ip6/::/tcp/4001'
+      ],
+      'API': '/ip4/127.0.0.1/tcp/5001',
+      'Gateway': '/ip4/127.0.0.1/tcp/8080'
+    },
+    'Discovery': {
+      'MDNS': {
+        'Enabled': true,
+        'Interval': 10
+      }
+    },
+    'Mounts': {
+      'IPFS': '/ipfs',
+      'IPNS': '/ipns'
+    },
+    'Ipns': {
+      'ResolveCacheSize': 128
+    },
+    'Gateway': {
+      'RootRedirect': '',
+      'Writable': false
+    },
+    'Bootstrap': [
+      '/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ',
+      '/ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z',
+      '/ip4/104.236.179.241/tcp/4001/ipfs/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM',
+      '/ip4/162.243.248.213/tcp/4001/ipfs/QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm',
+      '/ip4/128.199.219.111/tcp/4001/ipfs/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu',
+      '/ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64',
+      '/ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd',
+      '/ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3',
+      '/ip4/104.236.151.122/tcp/4001/ipfs/QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx'
+    ]
+  }
+
+  // Verify repo does not yet exist (or that 'force' is provided).
+  repo.exists((err, res) => {
+    if (err) { return callback(err) }
+    if (res === true && !opts.force) {
+      return callback(new Error('repo already exists and \'force\' is not set'))
+    }
+
+    generateAndSetKeypair()
+  })
+
+  // Generate peer identity keypair + transform to desired format + add to config.
+  function generateAndSetKeypair () {
+    var keys = peerId.create({
+      bits: opts.bits
+    })
+    config.Identity = {
+      PeerID: keys.toB58String(),
+      PrivKey: keys.privKey.toString('base64')
+    }
+
+    writeVersion()
+  }
+
+  function writeVersion () {
+    var version = require('../../package.json').version
```

Comment:
the 'repo version' is actually the 'repo version', not the IPFS implementation version. In js-ipfs, we use the latest repo available for go-ipfs, which is version 3. For more info, check here - https://github.com/ipfs/fs-repo-migrations